packages/template-retail-react-app/app/hooks/use-datacloud.js

```diff
@@ -0,0 +1,479 @@
+/*
+ * Copyright (c) 2025, Salesforce, Inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+import {useMemo} from 'react'
+import logger from '@salesforce/retail-react-app/app/utils/logger-instance'
+import {initDataCloudSdk} from '@salesforce/cc-datacloud-typescript'
+import {getConfig} from '@salesforce/pwa-kit-runtime/utils/ssr-config'
+import {useUsid, useCustomerId, useCustomerType, useDNT} from '@salesforce/commerce-sdk-react'
+import useMultiSite from '@salesforce/retail-react-app/app/hooks/use-multi-site'
+import {useCurrentCustomer} from '@salesforce/retail-react-app/app/hooks/use-current-customer'
+import {identity} from 'lodash'
+
+export class DataCloudApi {
+    constructor({site, appSourceId, tenantId, dnt}) {
+        this.site = site
```

Comment:
In pwa-kit, when we say `site`, it is refering to a site object. {id: 'siteId', l10n: {}}. I notices you are using it as site id. We should make it more cleaner by calling it `siteId` here
```suggestion
        this.siteId = siteId
```